pandas/tests/frame/test_arithmetic.py

```diff
@@ -2192,3 +2173,63 @@ def test_mixed_col_index_dtype(string_dtype_no_object):
     expected.columns = expected.columns.astype(string_dtype_no_object)
 
     tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("op", ["add", "sub", "mul", "div", "mod", "truediv", "pow"])
+def test_df_fill_value_operations(op):
+    # GH 61581
+    input_data = np.arange(50).reshape(10, 5)
+    fill_val = 5
+    columns = list("ABCDE")
+    df = DataFrame(input_data, columns=columns)
+    for i in range(5):
+        df.iat[i, i] = np.nan
+        df.iat[i + 1, i] = np.nan
+        df.iat[i + 4, i] = np.nan
+
+    df_base = df.iloc[:, :-1]
+    df_mult = df.iloc[:, -1]
+    mask = df.isna().values
+    mask = mask[:, :-1] & mask[:, [-1]]
+
+    df_result = getattr(df_base, op)(df_mult, axis=0, fill_value=fill_val)
+    df_expected = getattr(df_base.fillna(fill_val), op)(
+        df_mult.fillna(fill_val), axis=0
+    ).mask(mask, np.nan)
+
+    tm.assert_frame_equal(df_result, df_expected)
+
+
+dt_params = [
+    (tm.ALL_INT_NUMPY_DTYPES, 5),
+    (tm.ALL_INT_EA_DTYPES, 5),
+    (tm.FLOAT_NUMPY_DTYPES, 4.9),
+    (tm.FLOAT_EA_DTYPES, 4.9),
+]
+
+dt_param_flat = [(dt, val) for lst, val in dt_params for dt in lst]
+
+
+@pytest.mark.parametrize("data_type, fill_val", dt_param_flat)
+def test_df_fill_value_dtype(data_type, fill_val):
+    # GH 61581
+    base_data = np.arange(50).reshape(10, 5)
+    df_data = pd.array(base_data, dtype=data_type)
+    columns = list("ABCDE")
+    df = DataFrame(df_data, columns=columns)
+    for i in range(5):
+        df.iat[i, i] = np.nan
+        df.iat[i + 1, i] = pd.NA
+        df.iat[i + 4, i] = pd.NA
```

Comment:
there's a lot going on here.  is this the simplest example you can come up with that hits the relevant issue?